experimental/dds/tree2/src/shared-tree-core/branch.ts

```diff
@@ -276,16 +276,17 @@ export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> exten
 	}
 
 	/**
-	 * Undoes the last completed transaction made by the client.
-	 * It is invalid to call it while a transaction is open (this will be supported in the future).
+	 * Undoes the last change made by the client. If there is no change to undo, or if there was no
+	 * {@link UndoRedoManager} provider when this branch was constructed, this method has no effect.
```

Comment:
Another option is to fail if `undo` is called when this branch has no URM. I'm not sure which policy is better. In a different world, the "revertible" branch could actually be a separate type (e.g. subclass) than the non-revertible branch, and then there would be no undo/redo methods available on the non-revertible one. However, this is very difficult (impossible?) to do in a clean way since branches are self-referential (they rebase and merge onto their own type, as well as fork to produce their own type) and it ends up having a lot of implications on how you can make subclasses or compositions.

Reply:
Since you also think it seems likely to indicate a bug, I'll make it fail. If we change our mind, it's much easier to "unfail" later than to introduce a new fail!